### PR TITLE
Clean up the watermark underscores

### PR DESCRIPTION
**Pull request by ThislsAUsername**
I guess this is why I didn't add symbol support before, heh.

### CODE REVIEW

**Comment by ThislsAUsername** [comment]
Commit #2 there adds the ability to reasonably pick a "canonical" CO portrait for any variants to use.
I noticed DS Lash's image took precedence over AW2's, and I didn't like it.

File: src/UI/UIUtils.java [hunk 1]
```diff
@@ -335,15 +335,18 @@ public static COSpriteSpec fromDisk(String col, String fac)
   public static final COSpriteSpec LA = COSpriteSpec.fromDisk("Lazurian"      , "Army");
   public static final COSpriteSpec IDS = COSpriteSpec.fromDisk("IDS"          , "Army");
 
+  public static COSpriteSpec[] CANON_FACTIONS;
+  public static SourceGames[] GAME_RESOLUTION_ORDER;
+
   public enum SourceGames
```
Comment (ThatIsAUsername) [inline src/UI/UIUtils.java:341]: Is it just me or is it strange to have the enum type plural?

Like, if I wanted to create an enum for the type of vehicle someone had, I wouldn't name it `VehicleTypes`, and then enumerate trains, planes, an automobiles, I'd call it `VehicleType`.

File: src/UI/PlayerSetupCommanderController.java [hunk 2]
```diff
@@ -33,10 +30,7 @@ public PlayerSetupCommanderController(ArrayList<CommanderInfo> infos, PlayerSetu
     myPlayerInfo = playerInfo;
     shouldSelectMultiCO = tagMode.supportsMultiCmdrSelect;
 
-    HashSet<COSpriteSpec> uniqueFactions = new HashSet<>();
-    for( int co = 0; co < cmdrInfos.size(); ++co )
-      uniqueFactions.add(cmdrInfos.get(co).baseFaction);
-    canonFactions = uniqueFactions.toArray(new COSpriteSpec[0]);
+    var canonFactions = UIUtils.CANON_FACTIONS;
```
Comment (ThatIsAUsername) [inline src/UI/PlayerSetupCommanderController.java:33]: Is it really so hard to say `CoSpriteSpec[]`? What's the advantage for using `var`?

File: src/UI/UIUtils.java [hunk 3]
```diff
@@ -335,15 +335,18 @@ public static COSpriteSpec fromDisk(String col, String fac)
...
   {
-    // In order of CO name discriminator resolution
-    AW3(GE,  "3"),
-    AW2(BM,  "2"),
+    // In display order, because programming is hard™
```
Comment (ThatIsAUsername) [inline src/UI/UIUtils.java:343]: Comment syntax error: unmatched quotation mark.

Comment (ThatIsAUsername) [reply src/UI/UIUtils.java:343]: Also I'm not sure I under stand the comment. This is the order in which we check faction portraits if the explicit one is not found. It doesn't line up with the `GAME_RESOLUTION_ORDER` below, which is what actually dictates the order in which the games are displayed on the CO selection screen.

Comment (ThatIsAUsername) [reply src/UI/UIUtils.java:343]: No that's backward. `GAME_RESOLUTION_ORDER` is used for the faction portraits. I'm not sure what the `SourceGames` ordering dictates, but the two aren't ordered the same.

Also it seems odd to have a separate enum array just to reorder the enum values.

File: src/UI/UIUtils.java [hunk 4]
```diff
@@ -352,6 +355,25 @@ public enum SourceGames
...
+          BW, NW, LA, };
+      // In order of CO name discriminator resolution
+      GAME_RESOLUTION_ORDER = new SourceGames[]
+        {
+            SourceGames.DEFEND_PEACE,
+            SourceGames.AW3,
+            SourceGames.AW2,
+            SourceGames.AW1,
+            SourceGames.AW4,
+            SourceGames.AWBW,
```
Comment (ThatIsAUsername) [inline src/UI/UIUtils.java:374]: 3 2 1 4?

**Comment by ThatIsAUsername** [comment]
Also, this is mostly unrelated to this PR, but I just noticed a fair number of CO images have the execute bit turned on:
```
ls -l res/co/ | grep rwx
-rwxrwxr-x 1 chris chris   971 Dec  1  2023 Brenner_eyes.png
-rwxrwxr-x 1 chris chris  1832 Dec  1  2023 Brenner_face.png
-rwxrwxr-x 1 chris chris 28176 Dec  1  2023 Brenner.png
-rwxrwxr-x 1 chris chris   524 Mar 17 13:22 Drake_3_eyes.png
-rwxrwxr-x 1 chris chris   768 Mar 17 13:22 Drake_3_face.png
-rwxrwxr-x 1 chris chris  4159 Mar 17 13:22 Drake_3.png
-rwxrwxr-x 1 chris chris   869 Mar 17 13:22 Lin_eyes.png
-rwxrwxr-x 1 chris chris  1747 Mar 17 13:22 Lin_face.png
-rwxrwxr-x 1 chris chris 14551 Mar 17 13:22 Lin.png
-rwxrwxr-x 1 chris chris   610 Mar 17 13:22 Olaf_3_eyes.png
-rwxrwxr-x 1 chris chris   904 Mar 17 13:22 Olaf_3_face.png
-rwxrwxr-x 1 chris chris  3942 Mar 17 13:22 Olaf_3.png
-rwxrwxr-x 1 chris chris  1069 Mar 17 13:22 Penny_eyes.png
-rwxrwxr-x 1 chris chris  2387 Mar 17 13:22 Penny_face.png
-rwxrwxr-x 1 chris chris 11522 Mar 17 13:22 Penny.png
-rwxrwxr-x 1 chris chris   576 Mar 17 13:22 Rachel_eyes.png
-rwxrwxr-x 1 chris chris   977 Mar 17 13:22 Rachel_face.png
-rwxrwxr-x 1 chris chris  4040 Mar 17 13:22 Rachel.png
-rwxrwxr-x 1 chris chris   574 Mar 17 13:22 Von Bolt_eyes.png
-rwxrwxr-x 1 chris chris   872 Mar 17 13:22 Von Bolt_face.png
-rwxrwxr-x 1 chris chris  4894 Mar 17 13:22 Von Bolt.png
```

File: src/UI/UIUtils.java (hunk 1 above)
Comment (ThislsAUsername) [reply src/UI/UIUtils.java:341]: I didn't know that was a rule.
My hands just type words and then there's code.

File: src/UI/PlayerSetupCommanderController.java (hunk 2 above)
Comment (ThislsAUsername) [reply src/UI/PlayerSetupCommanderController.java:33]: Less typing.
I can make it explicit if you think `UIUtils.CANON_FACTIONS` is too ambiguous re: type.

File: src/UI/UIUtils.java (hunk 3 above)
Comment (ThislsAUsername) [reply src/UI/UIUtils.java:343]: > Comment syntax error: unmatched quotation mark.

That's not a quotation mark.
> No that's backward.

Yes.
> `GAME_RESOLUTION_ORDER` is used for the faction portraits.

Yes. I guess I could call it `PORTRAIT_RESOLUTION_ORDER`, but that's even longer.
> I'm not sure what the `SourceGames` ordering dictates, but the two aren't ordered the same.

Display order on the CO selection screen.
> Also it seems odd to have a separate enum array just to reorder the enum values.

If you have a better solution, I'm all ears.
I thought about giving the enum values a portrait precedence value or something, but that's less clear and seems less efficient (need to sort the enums by a different ordering, or iterate over all of them).

File: src/UI/UIUtils.java (hunk 4 above)
Comment (ThislsAUsername) [reply src/UI/UIUtils.java:374]: That's what it was before.
The ordering is like:
- Canonical
- Newest Trilogy entry
- Older
- Oldest
- One of these things is not like the others.

It wouldn't hurt anything to put DoR first, I suppose.

**Comment by ThislsAUsername** [comment]
> Also, this is mostly unrelated to this PR, but I just noticed a fair number of CO images have the execute bit turned on:

I blame Jay.

File: src/UI/UIUtils.java (hunk 3 above)
Comment (ThatIsAUsername) [reply src/UI/UIUtils.java:343]: > That's not a quotation mark.

Oh, right. Guess I shoulda looked twice.

> I guess I could call it PORTRAIT_RESOLUTION_ORDER, but that's even longer.

Also more specific to how it's used.

> If you have a better solution, I'm all ears.

Given that it's preference-driven, and your preference for CO-selection-screen ordering is different from your preference for portrait ambiguity resolution, I think this is as good as anything.

**Comment by ThatIsAUsername** [comment]
This is ready I think, unless you want to unpick some nits.

**Comment by ThislsAUsername** [comment]
Do you feel strongly about any of them?
I don't super care, but I don't mind changing them.

**Comment by ThatIsAUsername** [comment]
> strongly

Nope.

**Pull request merged**